sources/packages/backend/libs/test-utils/src/factories/application.ts

```diff
@@ -48,7 +48,7 @@ export function createFakeApplication(
   application.data = options?.initialValue.data ?? ({} as ApplicationData);
   application.programYear = relations?.programYear ?? createFakeProgramYear();
   // TODO get programYear from relations instead of setting the id here.
-  application.programYear.id = 2;
+  application.programYear.id = relations?.programYear?.id ?? 2;
```

Comment:
The lines 49 and 51 do not make sense but changing it now can impact the other E2E tests.
I will consider it not part of the PR.